### ISSUE TITLE
Expose function to find the hash method of a signature

### ISSUE THREAD

**Issue by jkrebs** [sybrenstuvel/python-rsa#78]
It would be great to expose a function that returns the hash method used in a signature. For example, we would like to enforce our customer's use of strong hash methods.

From https://github.com/sybrenstuvel/python-rsa/blob/master/rsa/pkcs1.py#L300:

```
def find_method_hash(signature, pub_key):
    keylength = common.byte_size(pub_key.n)
    encrypted = transform.bytes2int(signature)
    decrypted = core.decrypt_int(encrypted, pub_key.e, pub_key.n)
    clearsig = transform.int2bytes(decrypted, keylength)

    return _find_method_hash(clearsig)
```

**Comment by sybrenstuvel** [comment]
Well, now we return `True`, which is silly, so I think this is a good idea.

**Comment by sybrenstuvel** [comment]
Will be released in version 4.0
